### PR TITLE
Make LSP work again!

### LINKED ISSUE #335: Jlinked Lsp failed to initialize
image

Why

### LINKED ISSUE #341: Fields related LSP functions are broken after moving field resolving to Resolver
image

### PR DESCRIPTION
**Pull request by imkiva**
fixed #335 
fixed #341

### CODE REVIEW

**Comment by imkiva** [comment]
Could @ice1000 please try whether https://github.com/aya-prover/aya-dev/pull/342/commits/e9dd3e0b7ec522063f461f05d7440c07fdce641a fixed [the jlink issue](#335 ) on Windows?

File: lsp/build.gradle.kts [hunk 1]
```diff
@@ -29,6 +29,7 @@ jlink {
   addExtraDependencies("jline-terminal-jansi")
   mergedModule {
     additive = true
+    requires("com.google.gson")
```
Comment (ice1000) [inline lsp/build.gradle.kts:32]: 🤯

File: base/src/main/java/org/aya/distill/ConcreteDistiller.java [hunk 2]
```diff
@@ -115,7 +115,7 @@ yield visitCalls(infix,
       case Expr.NewExpr expr -> Doc.cblock(
         Doc.sep(Doc.styled(KEYWORD, "new"), term(Outer.Free, expr.struct())),
         2, Doc.vcat(expr.fields().view().map(t ->
-          Doc.sep(Doc.symbol("|"), Doc.plain(t.name()),
+          Doc.sep(Doc.symbol("|"), Doc.plain(t.name().data()),
```
Comment (ice1000) [inline base/src/main/java/org/aya/distill/ConcreteDistiller.java:118]:
```suggestion
          Doc.sep(Doc.symbol("|"), Doc.styled(FIELD, t.name().data())),
```

**Review by ice1000 — COMMENTED** [review]
Boo

File: lsp/build.gradle.kts (hunk 1 above)
Comment (imkiva) [reply lsp/build.gradle.kts:32]: Guessed from the error message. Passed LSP initialization in my machine 😆

**Comment by ice1000** [comment]
> Could @ice1000 please try whether [e9dd3e0](https://github.com/aya-prover/aya-dev/commit/e9dd3e0b7ec522063f461f05d7440c07fdce641a) fixed the jlink issue(#335) on Windows?

Could, tested and confirmed. Your next line is:
bors merge

**Pull request merged**